### ISSUE TITLE
Default operator suggesting the wrong type in every DependencyProperty

### ISSUE THREAD

**Issue by Sergio0694** [dotnet/csharplang#1150]
Hello, since I've upgraded to C# 7.1, I've noticed that VS (VS Enterprise 2017, version 15.4.4) started suggesting to use the new `default` literal in places where it would return the wrong type.

As an example, consider this simple `DependencyProperty`:

```csharp
public static readonly DependencyProperty CompactPaneLengthProperty = DependencyProperty.Register(
        nameof(CompactPaneLength), 
        typeof(double), 
        typeof(SlidingSplitView), 
        new PropertyMetadata(default(double), OnCompactPaneLengthPropertyChanged));
```

Now, the property exposes a `double` type and I'm initializing it with `default(double)` (so the value 0), but VS keeps underlining that and suggesting to just use `default`.

**But**, since the first parameter of the constructor for `PropertyMetadata` is an `object`, using just `default` would case it to be initialized with `null`, which is *not* what I want here (in fact, doing so causes a crash in my situation).

So my question is:

> Shouldn't VS *only* suggest the new `default` literal expression when it's actually sure it'll return the same type as the expression currently in use?

Is this the expected behavior? Thanks!

![The current VS suggestion](https://user-images.githubusercontent.com/10199417/33255834-a5387fd2-d34f-11e7-808f-f02f365dae0e.png)

**Comment by MkazemAkhgary** [comment]
same behavior. more like IgnoraSense than IntelliSense

    object x = default(double);

**Comment by Joe4evr** [comment]
This should be reported on [dotnet/roslyn](https://github.com/dotnet/roslyn/issues).

**Comment by ufcpp** [comment]
This has been already reported on the roslyn repository and will be fixed in VS 15.5.

**Comment by sharwell** [comment]
Duplicate of dotnet/roslyn#22184.
